cmd/minikube/cmd/docker-env.go

```diff
@@ -172,6 +182,21 @@ var dockerEnvCmd = &cobra.Command{
 			}
 		}
 
+		out, err := tryDockerConnectivity("docker", ec)
+		if err != nil { // docker might be up but been loaded with wrong certs/config
+			if strings.Contains(err.Error(), "x509: certificate is valid") {
+				glog.Infof("dockerd inside minkube is loaded with old certs with wrong IP. output: %s error: %v", string(out), err)
+			} else {
+				glog.Warningf("couldn't connect to docker inside minikube. output: %s error: %v", string(out), err)
+			}
+			// on minikube stop, or computer restart the IP might change.
+			// reloads the certs to prevent #8185
+			glog.Infof("will try to restart dockerd service...")
+			restartOrExitDaemon("docker", cname, co.CP.Runner)
+			// TODO: use kverify to wait for apisefver instead #8241
```

Comment:
I don't understand this TODO. The apiserver should not be a requirement to push images to the Docker daemon. Otherwise docker push won't work on paused clusters.

Reply:
this was to pass the integraiton tests in functional test. because right after docker-env we do other tests we wanna give back a healthy cluster.